channels/soundcloudChannel.py

```diff
@@ -0,0 +1,64 @@
+import random
+import struct
+import wave
+import soundcloud
+import os, re, urllib
+
+requiredParams = {
+        'sending': {
+            'ID':'Application ID for Soundcloud API',
+            'secret': 'Application secret for Soundcloud API',
+            'username': 'Username of Soundcloud account to post data to',
+            'password': 'Password of Soundcloud account to post data to',
+            'song_name': 'Name to be shown for "song" on soundcloud'
+            },
+        'receiving': {
+            'username': 'Username for user to download sound from.',
+            'song_name':'Name of the sound file to be downloaded'
+            }
+        }
+
+def send(data, params):
+    client = soundcloud.Client(
+        client_id=params['ID'],
+        client_secret=params['secret'],
+        username=params['username'],
+        password=params['password']
+    )
+
+    frames = []
+
+    for i in data:
+        frames.append(i)
+        frames.append(',')
```

Comment:
Does there need to be any sanity checking on the frames? If I just ran some data through b64 and passed it to this, would it work?

If not, just do some quick sanity checking and raise an [appropriate exception](https://docs.python.org/2/library/exceptions.html) if something's wrong. (Better yet, make it so it'll be compatible - if that's something that can be done.)

I just want to make sure that module linkages (i.e. *this* encoder must be run before *that* module) are as few as we can make them, and that they're really obvious when they happen.